src/main/kotlin/io/kuzzle/sdk/coreClasses/http/Route.kt

```diff
@@ -138,11 +146,20 @@ class Route {
         return HttpRequest(
             verb,
             if (queryArgs.isEmpty()) urlBuilder.toString() else "$urlBuilder?$queryString",
-            if (verb != "GET") request.optMap("body", KuzzleMap()) else null,
+            if (verb != "GET") getBody(request) else null,
             headers
         )
     }
 
+    private fun getBody(request: KuzzleMap): KuzzleMap {

```

Comment:
Body might not be a KuzzleMap and could be a RawJson / Serializable Object, so you should apply the same logic as in my previous comment